bin/propolis-server/src/lib/migrate/source.rs

```diff
@@ -60,6 +60,18 @@ struct SourceProtocol<T: AsyncRead + AsyncWrite + Unpin + Send> {
 
     /// Transport to the destination Instance.
     conn: WebSocketStream<T>,
+
+    // TODO: A hacky way for now to hang onto the VMM timing data between
+    // migration phases.
+    //
+    // We want to read the VMM timing data as soon as we can after we pause the
```

Comment:
> We want to read the VMM timing data as soon as we can after we pause the
> source, and make adjustments on the destination as close as we can to
> the end of the migration.

With the way things are structured, I'm not sure that's necessary for accuracy?

As Greg notes, we could change some of the ordering around when userspace device entities are paused vs the vCPUs.  I imagine we'll end up evolving those events with more time (such as adding a stage when disk/network IO is throttled, but the devices themselves aren't paused).

Reply:
this is all restructured as of cdbe3ef, though I still kept around a field to hang on to time data state between export and send in the protocol

Reply:
as of d09938fb, there is only one time data phase of the protocol

I realized through review and discussion that having two phases more closely mapped to my initial model of how to correct the guest TSC (namely, measure how long migration takes and add that to the guest TSC). But it doesn't actually matter for correctness when we export/import the data, since the guest TSC is calculated from host hrtime. As long as we import the time data before the device state, we should be in good shape.